frontend/src/scenes/events/EventDetails.js

```diff
@@ -1,17 +1,25 @@
-import React from 'react'
+import React, { useState } from 'react'
 import { PropertiesTable } from 'lib/components/PropertiesTable'
 import moment from 'moment'
 import { EventElements } from 'scenes/events/EventElements'
 import { Tabs, Button } from 'antd'
 
 import { createActionFromEvent } from './createActionFromEvent'
+import { keyMapping } from 'lib/components/PropertyKeyInfo'
 const { TabPane } = Tabs
 
 export function EventDetails({ event }) {
-    let displayableEventProperties = {}
+    const [showHiddenProps, setShowHiddenProps] = useState(false)
 
+    let displayableEventProperties = {}
+    let hiddenPropsCount = 0
     for (let key of Object.keys(event.properties)) {
-        if (key !== '$snapshot_data') displayableEventProperties[key] = event.properties[key]
+        if (keyMapping.event[key] && keyMapping.event[key].hide) {
```

Comment:
Q: I think lodash would clean things up a lot here, would it make sense to introduce it?

```js
import { omit, pick, keys } from 'lodash' 

const hideableKeys = pick(keys(event.properties), (key) => keyMapping[key].hide)
const displayedProperties = showHiddenProps ? event.properties : omit(event.properties, hideableKeys)
```